### PR TITLE
CLN: enforce deprecation of the `kind` keyword on df.resample/ser.resample

### PR DESCRIPTION
**Pull request by natmokval**
xref #55895
enforced deprecation of the kind keyword on df.resample/ser.resample

### CODE REVIEW

**Comment by natmokval** [comment]
I enforced deprecation of the `kind` keyword on `df.resample/ser.resample`.
@jbrockmendel, could you please take a look at this PR?

File: pandas/tests/resample/test_datetime_index.py [hunk 1]
```diff
@@ -449,9 +449,7 @@ def test_resample_frame_basic_kind(freq, unit):
         index=date_range("2000-01-01", periods=10, freq="B"),
     )
     df.index = df.index.as_unit(unit)
-    msg = "The 'kind' keyword in DataFrame.resample is deprecated"
-    with tm.assert_produces_warning(FutureWarning, match=msg):
-        df.resample(freq, kind="period").mean()
```
Comment (jbrockmendel) [inline pandas/tests/resample/test_datetime_index.py:454]: i think you can remove this test entirely

File: pandas/core/generic.py [hunk 2]
```diff
@@ -8976,7 +8955,6 @@ def resample(
             freq=rule,
             label=label,
             closed=closed,
-            kind=kind,
```
Comment (jbrockmendel) [inline pandas/core/generic.py:8979]: i dont see a diff in pandas.core.resample; doesn't the keyword need to be removed there too?

**Comment by jbrockmendel** [comment]
@ChadFulton any objection to moving forward with this part of the deprecation?

**Comment by ChadFulton** [comment]
Thanks @jbrockmendel, this change makes a lot of sense to me!

File: pandas/core/generic.py (hunk 2 above)
Comment (natmokval) [reply pandas/core/generic.py:8979]: sorry, my mistake. I removed the keyword `kind` from `get_resampler`

File: pandas/tests/resample/test_datetime_index.py (hunk 1 above)
Comment (natmokval) [reply pandas/tests/resample/test_datetime_index.py:454]: thanks, done

File: pandas/core/generic.py (hunk 2 above)
Comment (mroeschke) [reply pandas/core/generic.py:8979]: Shouldn't all/most of the `kind` keywords in `resample.py` be removable?

Comment (natmokval) [reply pandas/core/generic.py:8979]: Thanks for the comment. Yes, it was my mistake; `kind` can be removed from `Resampler` and `TimeGrouper`. I have updated this PR. Could you please take a look?

File: pandas/tests/resample/test_period_index.py [hunk 3]
```diff
@@ -489,16 +479,17 @@ def test_cant_fill_missing_dups(self):
             s.resample("Y").ffill()
 
     @pytest.mark.parametrize("freq", ["5min"])
```
Comment (mroeschke) [inline pandas/tests/resample/test_period_index.py:481]: Could you inline this parameter since it's the only one?

**Review by mroeschke — COMMENTED** [review]
One comment otherwise LGTM

File: pandas/tests/resample/test_period_index.py (hunk 3 above)
Comment (natmokval) [reply pandas/tests/resample/test_period_index.py]: done

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @natmokval

**Pull request merged**